server/src/graphql/resolvers/deliveryPhase.js

```diff
@@ -0,0 +1,42 @@
+const { db } = require('../../db/models');
+
+const groupByType = (parent) => {
+  return parent.reduce((result, currentValue) => {
+    (result[currentValue.typeId] = result[currentValue.typeId] || []).push(
+      currentValue
+    );
+    return result;
+  }, {});
+};
+
+// schema will not accept spaces or special characters as type field names so we must replace versus a lookup (which would return strings with spaces and special characters)
```

Comment:
Can you not just make a copy of the object without the hidden attributes, and use the copy to update?